README.md

```diff
@@ -22,44 +23,106 @@ The main purpose of the library is to handle 4 key functions:
 4. Maintaining and providing logs from a jobs output.
 
 ``` go
+type JobStatus struct {
+	// Options - ["Stopped", "Completed", "Errored"]
+	status string
+	exitCode uint8
+	output JobOutput
+}
+
+type JobOutput struct {
+	StdOut []byte
+	StdErr []byte
+}
+
+type JobsInfo struct {
+	Command *exec.Cmd
+	JobStatus JobStatus
+}
+
+// Note: All channels will be buffered with cap 1.
+type JobChans struct {
+	// Write from API
+	Kill chan struct{}{}
+	// Read from API
+	Status chan JobStatus
+}
+
 type JobsManager struct {
-	Jobs sync.Map
-
-	// These fields are used for pseduo-persistence.
-	// Type: uuid.UUID -> []byte
-	StdOut sync.Map
-	StdErr sync.Map 
-
-	// We store status locally because PIDs can cycle.
-	// This could cause integerity concerns.
-	// e.g Job A with PID 1 completes but Job B cycles back to PID 1
-	Status sync.Map
-	ExitCode sync.Map
+	JobInfo sync.Map // uuid -> JobsInfo
+	JobChannels sync.Map
 }
 ```
 
 ## Start
 ```go
-func (*JobsManager) Start(string, ...string) (uuid.UUID, bool)
+func (*JobsManager) Start(cmd string, args ...string) (uuid.UUID)
 ```
 There are two major parts to this function.
 1. Construct the command object with the parameters provided.
-2. Spin up a goroutine, which will manage mutation to State variables (e.g Status, ExitCode, Std(out|err)).
+2. Spin up a goroutine, which will send info through status channel when complete.
 
-Finally the function should return the UUIDv4 and a boolean signature notifying success of job creation.
+Finally the function should return the UUIDv4.
 
 ## Query
 ``` go
-func (*JobsManager) Query(uuid.UUID) (exited bool, exitCode int, error)
+func (jm, *JobsManager) Query(uuid.UUID) (found bool, js JobStatus)
+```
+This function will use the uuid to `Load` job status data structure stored inside the jobs Map. If not found in JobsInfo, it will return False. It will detect "active" jobs via a select statments.
+
+```go
+// Check channels for value, if exist, load into cache/sync map.
+chans := jm.JobChannels.Load(uuid)
+select {
+	case status <-chans.status
+		// Load map with status.
+	default:
+		continue
+}
+...
+jobInfo = jm.JobInfo.Load(uuid)
+if jobInfo.Command == nil {
+	return false, nil
+}
+else if jobInfo.JobStatus == nil {
+	return true, JobStatus{status:"Active"}
+} else{
+	return true, jobInfo.JobStatus
+}
 ```
-This function will use the uuid to `Load` the correct values from the various sync.Maps associated with job state and output.
 
 ## Stop
 ``` go
-func (*JobsManager) Stop(uuid.UUID) (bool, error)
+func (*JobsManager) Stop(uuid.UUID) (killSent bool, error)
```

Comment:
Why does this need to return `killSent`? What would be the case when it's `false`?

Reply:
The case where the job had already completed and the Kill Signal could not be sent.
- It really doesn't need to be there as long as the function provides the contract that the system has made a best effort to close any running jobs.